torch/_inductor/scheduler.py

```diff
@@ -1724,17 +1702,18 @@ def add_user(
                     # this node must run after the prior writer
                     add_user(alt_name, node)
                     node.add_fake_dep(StarDep(alt_name, mode=node_mode))
-                    known_dep_node_names = dep_closure(node)
                     for user in name_to_users[alt_name].items:
+                        if user.get_name() == node.get_name():
```

Comment:
Should a node ever be a user of itself?

Reply:
`alt_name` here should be the pre-mutation name for the bufer, so `x.add_(1)` would be a user of `x` and have a mutation on the renamed `x` buffer.